docs/proposals/conformance-test.md

```diff
@@ -0,0 +1,147 @@
+- The tests should have a well documented and short list of set-up requirements.
+- The tests should install and complete in a relatively short period of time
+  with suggested minimum infrastructure requirements
+  (e.g. 3 nodes, 24 vCPU, 64 GB RAM, 500 GB Disk).
+
+## Kubeflow Conformance
+
+Initially the Kubeflow conformance will include the CRD based tests.
+In the future, API and UI based tests may be added. Kubeflow conformance consists
+the 3 category of tests:
+
+- CRD-based tests
+
+  Most of Katib and Training Operator functionality are based on Kubernetes CRD.
+
+  **This document will define a design for CRD-based tests for Katib and the Training Operator.**
+
+- API-based tests
+
+  Currently, Katib or Training Operator doesn’t have an API server that receives
+  requests from the users. However, Katib has the DB Manager component that is
+  responsible for writing/reading ML Training metrics.
+
+  In the following versions, we should design conformance program for the
+  Katib API-based tests.
+
+- UI-based tests
+
+  UI tests are valuable but complex to design, document and execute. In the following
+  versions, we should design conformance program for the Katib UI-based tests.
+
+## Design for the CRD-based tests
+
+![conformance-crd-test](../images/conformance-crd-test.png)
+
+The design is similar to the KFP conformance program for the API-based tests.
+
+For Katib, tests will be based on
+[the `run-e2e-experiment.go` script](https://github.com/kubeflow/katib/blob/570a3e68fff7b963889692d54ee1577fbf65e2ef/test/e2e/v1beta1/hack/gh-actions/run-e2e-experiment.go)
+that we run for our e2e tests.
+
+This script will be converted to use Katib SDK. Tracking issue: https://github.com/kubeflow/katib/issues/2024.
+
+For the Training Operator, tests will be based on [the SDK e2e test.](https://github.com/kubeflow/training-operator/tree/05badc6ee8a071400efe9019d8d60fc242818589/sdk/python/test/e2e)
+
+### Test Workflow
+
+All tests will be run in the _kf-conformance_ namespace inside the separate container.
+That will help to avoid environment variance and improve fault tolerance. Driver is required to trigger the deployment and download the results.
+
+- We are going to use
+  [the unified Makefile](https://github.com/kubeflow/kubeflow/blob/2fa0d3665234125aeb8cebe8fe44f0a5a50791c5/conformance/1.5/Makefile)
+  for all Kubeflow conformance tests. Distributions (_driver_ on the diagram)
+  need to run the following Makefile commands:
+
+  ```makefile
+
+  # Run the conformance program.
+  run: setup run-katib run-training-operator
+
+  # Sets up the Kubernetes resources (Kubeflow Profile, RBAC) that needs to run the test.
+  # Create temporary folder for the conformance report.
+  setup:
+    kubectl apply -f ./setup.yaml
+    mkdir -p /tmp/kf-conformance
+
+  # Create deployment and run the e2e tests for Katib and Training Operator.
+  run-katib:
+    kubectl apply -f ./katib-conformance.yaml
```

Comment:
How about creating as a k8s job to be completed if test success of not, rather than creating deployments and remove deployments?

Reply:
@anencore94 I guess, we want to use deployment to give user ability to download results after it succeeded.
So after test is finished, we should move container to busy-box state until user deletes the deployment.
Is that correct @james-jwu ?

Reply:
@james-jwu No, pod will stay in `Complete` state, if don't set the [ttl parameter for the Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/#ttl-mechanism-for-finished-jobs).
I guess, user can just verify the Job status to see if tests have been finished.

Reply:
We do need to generate a report that can be submitted as a PR. You can use a job as long as the report can be generated. Ideally the report is available any time before `make clean` is called.

Reply:
LGTM